foxai/explainer/computer_vision/algorithm/gradient_utils.py

```diff
@@ -0,0 +1,174 @@
+from inspect import signature
+from typing import Any, Callable, Optional, Tuple
+
+import torch
+
+from foxai.types import LayerType, ModelType, TargetType
+
+
+def _run_forward(
+    forward_func: Callable,
+    input_tensor: torch.Tensor,
+    target: Optional[TargetType] = None,
+    additional_forward_args: Optional[Tuple[Any]] = None,
+) -> torch.Tensor:
+    r"""Executes forward pass and returns result with respect to the target.
+
+    Args:
+        forward_func: forward function. This can be for example model's forward function.
+        input_tensor:      Input to be passed to forward_fn.
+        target: Index of the target class for which gradients must be computed (classification only).
+        additional_forward_args: Additional input arguments that forward function requires.
+    """
+    if len(signature(forward_func).parameters) == 0:
+        output = forward_func()
+    elif additional_forward_args is not None:
```

Comment:
I don't think we will use `additional_forward_args` in the near future. I think we could remove this argument to simplify the code and add it later when we will need it. What do You think @rpytel1 @adamjankaczmarek @kamilrzechowski?

Reply:
I think that's awesome idea. Just have in mind, all other algorithms relying on captum support this at the moment. However if we all agree I will gladly remove it.

Reply:
As discussed on project sync, we are leaving this argument.